contracts/aggregator/mainnet/flashloan/helpers.sol

```diff
@@ -7,14 +7,29 @@ import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
 import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
 
 import {
-    TokenInterface,
     CTokenInterface,
     CEthInterface
 } from "./interfaces.sol";
 
 contract Helper is Variables {
     using SafeERC20 for IERC20;
 
+    /**
+     * @dev Approves the token to the spender address with allowance amount.
+     * @notice Approves the token to the spender address with allowance amount.
+     * @param token_ token for which allowance is to be given.
+     * @param spender_ the address to which the allowance is to be given.
+     * @param amount_ amount of token.
+    */
+    function approve(IERC20 token_, address spender_, uint256 amount_) internal {
+        if(token_.allowance(address(this), spender_) == 0) {
```

Comment:
@Vaibhav20201 I think if we use by default this, it will increase the gas for users. Let's think around to reduce at least on mainnet?

Reply:
Okay

Reply:
```solidity
    function approve(IERC20 token, address spender, uint256 amount) internal {
        try token.approve(spender, amount) {

        } catch {
            try token.approve(spender, 0) {
               token.approve(spender, amount);
            } catch {
               token.safeApprove(spender, 0);
               token.safeApprove(spender, amount);
            }
            
        }
    }
```

Reply:
This code looks better.

Reply:
It fails on my end
![image](https://user-images.githubusercontent.com/55931792/147341141-f8d4e44e-2656-49dd-afc0-398a8a6825bb.png)

Reply:
Fixed